packages/mui-base/src/utils/useCompoundItem.ts

```diff
@@ -55,13 +57,13 @@ export function useCompoundItem<Key, Subitem>(
   }
 
   const { registerItem } = context;
-  const [registeredId, setRegisteredId] = React.useState(id);
+  const [registeredId, setRegisteredId] = React.useState(typeof id === 'function' ? undefined : id);
```

Comment:
since parameter to pass to the `id` function is not available at this point, i initiated value of `registeredId` to `undefined` if `id` is function.